src/ray/core_worker/core_worker.h

```diff
@@ -486,7 +480,7 @@ class CoreWorker {
   std::shared_ptr<gcs::RedisGcsClient> gcs_client_;
 
   // Client to the raylet shared by core worker interfaces.
-  std::unique_ptr<RayletClient> raylet_client_;
+  std::shared_ptr<RayletClient> raylet_client_;
```

Comment:
Could you explain on why this needs to be a shared_ptr? Thanks.

Reply:
It's because of the worker lease design. The raylet client implements the lease client interface. You can request multiple workers through the same lease client, so we need to make sure that we keep the connection alive while we're still leasing the workers. There are other ways to do it, but it seemed simplest to just use a shared_ptr to the lease client. I'll add a note about this to the header.

Reply:
got it, thanks.